eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -393,4 +403,21 @@ class EclairImpl(appKit: Kit) extends Eclair {
     val sendPayment = SendPaymentRequest(amount, paymentHash, recipientNodeId, maxAttempts, externalId = externalId_opt, routeParams = Some(routeParams), userCustomTlvs = keySendTlvRecords)
     (appKit.paymentInitiator ? sendPayment).mapTo[UUID]
   }
+
+  override def signMessage(base64Message: ByteVector, prefix: ByteVector): SignedMessage = {
+    val hash256Message = Crypto.hash256(prefix ++ base64Message)
+    val signature = appKit.nodeParams.keyManager.signDigest(hash256Message)
+    SignedMessage(appKit.nodeParams.keyManager.nodeId, base64Message.toBase64, signature)
+  }
+
+  override def verifyMessage(base64Message: ByteVector, signature: ByteVector64, prefix: ByteVector): VerifiedMessage = {
+    val hash256Message = Crypto.hash256(prefix ++ base64Message)
+    val (pubKey1, pubKey2) = Crypto.recoverPublicKey(signature, hash256Message)
+    if (appKit.nodeParams.db.network.getNode(pubKey1).isDefined)
```

Comment:
I don't think that's how we should do it, it feels hackish.
Thinking about this more, I think we should also include the `recoveryId` byte in our signature and use it (which allows us to use the `Crypto.recoverPublicKey` that takes a `recoveryId` and returns a single `PublicKey` (and internally uses `libsecp256k1` which is better than bouncycastle).
What do you think? I don't remember the exact details of how that `recoveryId` works but there's probably some documentation somewhere.